### PR TITLE
Fix "disable mouse buttons" not working

### LINKED ISSUE #2840: Mouseclick + F10 = Crash
When i hold Leftclick while playing and toggle off the mouse buttons via F10 the game crashes. It happens everytime.
This is just here to report this bugs, since i am not able to do it via the launcher.
I hope this will be patched sometime in the future, i know there is more important stuff to do

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/59864811-mouseclick-f10-crash?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### LINKED ISSUE #2868: Disabled mouse failure
OSU Lazer will still enable mouse after disabled mouse is enabled.

image
Version: 2018.622.0

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/60050605-disabled-mouse-failure?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ekrctb**
- [x] Depends on ppy/osu-framework#1682
- Resolves #2868, fixes #2840.

### CODE REVIEW

**Review by peppy — DISMISSED** [review]

File: osu.Game/Overlays/Settings/RulesetSettingsSubsection.cs [hunk 1]
```diff
@@ -25,9 +28,9 @@ protected override IReadOnlyDependencyContainer CreateChildDependencies(IReadOnl
         {
             dependencies = new DependencyContainer(base.CreateChildDependencies(parent));
 
-            var config = dependencies.Get<RulesetConfigCache>().GetConfigFor(ruleset);
-            if (config != null)
-                dependencies.Cache(config);
+            Config = dependencies.Get<RulesetConfigCache>().GetConfigFor(ruleset);
+            if (Config != null)
+                dependencies.CacheAs(Config);
```
Comment (smoogipoo) [inline osu.Game/Overlays/Settings/RulesetSettingsSubsection.cs]: Wait why was this changed to a CacheAs?

Comment (peppy) [reply osu.Game/Overlays/Settings/RulesetSettingsSubsection.cs]: If not it will be cached as an `IRulesetConfigManager`, which is less useful to children (would need a cast at each usage).

Comment (smoogipoo) [reply osu.Game/Overlays/Settings/RulesetSettingsSubsection.cs]: No, it's cached as an `IRulesetConfigManager` if you use `CacheAs`. `CacheAs` uses the type passed into the generic parameter (`IRulesetConfigManager` in this case).

Comment (peppy) [reply osu.Game/Overlays/Settings/RulesetSettingsSubsection.cs]: so it does. that method name is a bit confusing..

**Review by peppy — APPROVED** [review]
